### DISCUSSION TITLE
Add points to WFS Layers with QField

### DISCUSSION THREAD

**Discussion by stevenrizo** [opengisch/QField#5164] [Q&A]
Hello,

I've been looking into some of the data collection and form options for QField and QGIS, but our data is currently being hosted and displayed in another app. I can pull this data in through a WFS, and I can easily add data points in the desktop version of QGIS. However, I need to be able to use mobile data collection through QField with these WFS layers, but haven't had any luck with this. 

I've had no issues collecting data that has been set up in a geopackage, but the WFS layers appear to be locked as read-only and I haven't found any way to change this.

Has anyone found a way to work with this kind of setup, or any alternative solutions from a similar set up?

All help greatly apprecaited.

**Comment by CodePutz** [comment]
Bump.

Same issue here. WFS layers are presented fine in QGIS. Adding, editing and deleting of features via WFS works fine. However, QField only presents the features. Adding, editing and deleting of features does not work. 

Is QField still missing WFS-T support for editing/deleting WFS layers?

**Reply by nirvn** [reply]
@CodePutz ,we'll need some sample project to test this.

**Reply by CodePutz** [reply]
I'm not sure what you mean. Do you want read/write access to a WFS server for implementing and testing WFS-T support for QField?

Or are you asking for a copy of my project file for analysis? I'm using the same project file in QGIS and QField. Add/modify/delete of features via WFS works there in QGIS. I'm seeing POST requests to the WFS server on edits/deletes. However, QField does not make any POST requests at all, as if "WFS-write" capabilities are not present.

Hence the question to confirm if QField is supposed to support WFS write/POST transactions or if that is still on the wish/todo-list.

If you do need a test-bed for development of those features, I can probably set up a QGIS server with PostGIS backend and make that available to you.

**Reply by nirvn** [reply]
@CodePutz , I'm requesting a sample QGIS project file where WFS feature addition/editing works in QGIS but fails in QField. 

And yes, QField does support these operations. Hence why we'd need to dissect this issue with through a project and WFS server that fails.

We can continue this discussion via email if you are able to share a project but do not want to drop it in this public space. Reach out to me via mathieu@opengis.ch .

**Reply by coastalrocket** [reply]
I'm working on projects where there's a desire for edits to be sent back to a postgis database. We're not keen on exposing a postgis db on an external IP so have been looking at WFS-T as replacement. The downside is that if mobile coverage drops then, and this is my understanding, QField doesn't have the means to wait until connection is made before attempting again. If there were some means of holding back on WFS-T transactions until connection is made that would be wonderful.
